src/hostname/hostnamectl.c

```diff
@@ -280,9 +301,24 @@ static int set_hostname(int argc, char **argv, void *userdata) {
         }
 
         if (arg_static) {
-                r = set_simple_string(bus, "static hostname", "SetStaticHostname", hostname);
-                if (r < 0)
-                        return r;
+                if (!implicit) {
+                        r = set_simple_string(bus, "static hostname", "SetStaticHostname", hostname);
+                        if (r < 0)
+                                return r;
+                } else {
+                        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+
+                        r = set_simple_string_internal(bus, &error, "SetStaticHostname", hostname);
+                        if (r < 0) {
+                                if (sd_bus_error_has_names(&error,
+                                                           BUS_ERROR_FILE_IS_PROTECTED,
+                                                           BUS_ERROR_READ_ONLY_FILESYSTEM)) {
+                                        log_warning_errno(r, "Could not set static hostname, ignoring: %s", bus_error_message(&error, r));
+                                        show_hint = true;
```

Comment:
I think this shouldn't be done: for most people, setting of the static hostname is the only thing that is important. The fact that the file is protected is a very good reason to fail. I think it is fine to only warn if the pretty hostname cannot be set, but not here. The hint should be shown, but the error code needs to be propagated.

@poettering, wdyt?

Reply:
Actually, for consistency, I'd prefer to propagate the error in both cases. Users will mostly interact with the hostnamed api through the graphical setttings dialogues. The commandline api is for scripts and automatic provisioning and such, and there it is almost always better to simply propagate errors to the caller if there is any doubt.